presto-main/src/main/java/com/facebook/presto/split/SplitManager.java

```diff
@@ -54,20 +63,30 @@ public void removeConnectorSplitManager(ConnectorId connectorId)
         splitManagers.remove(connectorId);
     }
 
-    public SplitSource getSplits(Session session, TableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public SplitSource getSplits(Session session, TableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        ConnectorId connectorId = layout.getConnectorId();
+        ConnectorId connectorId = table.getConnectorId();
         ConnectorSplitManager splitManager = getConnectorSplitManager(connectorId);
 
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
+        // TODO add support in metadata to check Connector still needs TableLayoutHandle and method in ConnectorSplitManager to use connectorTableHandle to get split
+        ConnectorTableLayoutHandle layout;
+        if (!table.getLayout().isPresent()) {
+            Optional<TableLayoutResult> result = metadata.getLayout(session, table, Constraint.alwaysTrue(), Optional.empty());
+            checkArgument(result.isPresent(), format("must be able to obtain a layout from this connector %s", connectorId));
+            layout = result.get().getLayout().getHandle().getConnectorHandle();
+        }
+        else {
+            layout = table.getLayout().get();
+        }
 
         ConnectorSplitSource source = splitManager.getSplits(
-                layout.getTransactionHandle(),
+                table.getTransaction(),
                 connectorSession,
-                layout.getConnectorHandle(),
+                layout,
```

Comment:
I know that this is pain in the neck, but if we are migrating the engine interfaces, we should also migrate the connector interfaces. Otherwise it quickly becomes messy.